### PR TITLE
Fixes None dimension bug

### PR DESCRIPTION
**Pull request by Lynx1820**
Bug: #4646 

ML.Net considers empty dimensions invalid. However, the "None" dimension is valid in OnnxRuntime.

### CODE REVIEW

File: src/Microsoft.ML.OnnxTransformer/OnnxTypeParser.cs [hunk 1]
```diff
@@ -177,10 +177,12 @@ public static int GetDimValue(OnnxCSharpToProtoWrapper.TensorShapeProto.Types.Di
                     break;
                 case OnnxCSharpToProtoWrapper.TensorShapeProto.Types.Dimension.ValueOneofCase.DimParam:
                     // Variable-length dimension is translated to 0.
-                    value = 0;
+                    break;
+                case 0:
```
Comment (harishsk) [inline src/Microsoft.ML.OnnxTransformer/OnnxTypeParser.cs]: The switch statement is on an enum. Can you please change the case 0: to use ValueOneCase.None?

**Comment by dasMulli** [comment]
Quick question: Would an error message saying that ML.Net doesn't support `None` dimension be more actionable than calling the dimension "invalid" in the exception text? At least this indicates that the onnx model isn't faulty but simply that using it is unsupported..

**Comment by Lynx1820** [comment]
> Quick question: Would an error message saying that ML.Net doesn't support `None` dimension be more actionable than calling the dimension "invalid" in the exception text? At least this indicates that the onnx model isn't faulty but simply that using it is unsupported..

Actually, I added another case for the None value. ML.NET should be able to load the model without an exception.

**Comment by najeeb-kazmi** [comment]
@Lynx1820 please pull master so that this PR can skip the randomly failing tests. Then we can merge this.

Also, does this fix #4646 or is there more work for that bug?

**Pull request merged**